devicetypes/smartthings/zigbee-sound-sensor/zigbee-sound-sensor.groovy

```diff
@@ -0,0 +1,173 @@
+                            [value: 96, color: "#bc2323"]
+                    ])
+        }
+
+        main "sound"
+        details(["sound", "battery", "temperature"])
+    }
+}
+
+def installed() {
+    sendEvent(name: "sound", value: "not detected", displayed: false)
+}
+
+def parse(String description) {
+    log.debug "Parsing: '${description}'"
+    def map = zigbee.getEvent(description)
+
+    if(!map) {
+        if(description?.startsWith('zone status')) {
+            map = parseIasMessage(description)
+        } else {
+            map = parseAttrMessage(description)
+        }
+    }
+
+    log.debug "Parse returned ${map}"
+    def result = map ? createEvent(map) : [:]
+
+    if (description?.startsWith('enroll request')) {
+        def cmds = zigbee.enrollResponse()
+        log.debug "enroll response: ${cmds}"
+        result = cmds?.collect { new physicalgraph.device.HubAction(it)}
+    }
+    return result
+}
+
+private Map parseIasMessage(String description) {
+    ZoneStatus zs = zigbee.parseZoneStatus(description)
+    def result = [:]
+    if(zs.isAlarm1Set() || zs.isAlarm2Set()) {
+        result = getSoundDetectionResult("detected")
+    } else if(!zs.isTamperSet()) {
+        result = getSoundDetectionResult("not detected")
+    } else {
+        result = [displayed: true, descriptionText: "${device.displayName}'s case is opened"]
+    }
+
+    return result
+}
+
+private Map parseAttrMessage(description) {
+    def descMap = zigbee.parseDescriptionAsMap(description)
+    def map = [:]
+    if(descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.commandInt != 0x07 && descMap?.value) {
+        map = getBatteryPercentageResult(Integer.parseInt(descMap.value, 16))
+    } else if(descMap?.clusterInt == zigbee.TEMPERATURE_MEASUREMENT_CLUSTER && descMap.commandInt == 0x07) {
+        if (descMap.data[0] == "00") {
+            sendEvent(name: "checkInterval", value: 60 * 60, displayed: true, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+        } else {
+            log.warn "TEMP REPORTING CONFIG FAILED - error code: ${descMap.data[0]}"
+        }
+    } else if(descMap.clusterInt == 0x0020 && descMap.commandInt == 0x00) {
+        log.debug "Check in interval command received!"
+        sendEvent(name: "checkInterval", value: 60 * 60, displayed: true, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+    } else if (map.name == "temperature") {
+        if (tempOffset) {
+            map.value = (int) map.value + (int) tempOffset
+        }
+        map.descriptionText = temperatureScale == 'C' ? '${device.displayName} was ${value}°C' : '${device.displayName} was ${value}°F'
+        map.translatable = true
+    }
+
+    return map
+}
+
+private Map getBatteryPercentageResult(rawValue) {
+    def result = [:]
+    def volts = rawValue / 10
+    if (!(rawValue == 0 || rawValue == 255)) {
+        def minVolts = 2.2
+        def maxVolts = 3.0
```

Comment:
Are these the right voltage values?

Does this device support reporting its battery as a percentage instead of as raw voltage? See `zigbee-lock-without-codes`

Reply:
In my first version of this DTH I went for 2.3V for minimum voltage, as this was a value of *BatteryVoltageMinThreshold* in Power Cluster, but I've changed it to 2.2V, according to Ecolink's sample DTH we've received recently.
Regarding battery reporting. I've tried to measure with *BatteryPercentageRemaining*, but I've received unsupported attribute error.